src/vcpkg/portfileprovider.cpp

```diff
@@ -174,40 +149,30 @@ namespace vcpkg
                         return msg::format_error(msgPortDoesNotExist, msg::package_name = version_spec.port_name);
                     }
 
-                    auto maybe_path = ent->get()->get_version(version_spec.version);
-                    if (auto path = maybe_path.get())
+                    auto maybe_scfl = ent->get()->try_load_port(version_spec.version);
+                    if (auto scfl = maybe_scfl.get())
                     {
-                        auto maybe_scfl =
-                            Paragraphs::try_load_port_required(m_fs, version_spec.port_name, *path).maybe_scfl;
-                        if (auto scfl = maybe_scfl.get())
+                        auto scf_vspec = scfl->to_version_spec();
+                        if (scf_vspec == version_spec)
                         {
-                            auto scf_vspec = scfl->to_version_spec();
-                            if (scf_vspec == version_spec)
-                            {
-                                return std::move(*scfl);
-                            }
-                            else
-                            {
-                                return msg::format_error(msgVersionSpecMismatch,
-                                                         msg::path = path->port_directory,
-                                                         msg::expected_version = version_spec,
-                                                         msg::actual_version = scf_vspec);
-                            }
+                            return std::move(*scfl);
                         }
                         else
                         {
-                            return std::move(maybe_scfl)
-                                .error()
-                                .append_raw('\n')
-                                .append_raw(NotePrefix)
-                                .append(msgWhileLoadingPortVersion, msg::version_spec = version_spec)
-                                .append_raw('\n');
+                            get_global_metrics_collector().track_define(DefineMetric::VersioningErrorVersion);
```

Comment:
This DefineMetric will now fire in more places now; 'the version DB says the version is one way but upon inspection it's something else' seems more appropriate for it? Would be happy to just delete it too /cc @vicroms